common/src/main/java/xyz/wagyourtail/jsmacros/client/mixins/access/MixinScreen.java

```diff
@@ -24,30 +30,41 @@
 import xyz.wagyourtail.jsmacros.client.access.CustomClickEvent;
 import xyz.wagyourtail.jsmacros.client.access.IScreenInternal;
 import xyz.wagyourtail.jsmacros.client.api.helpers.ButtonWidgetHelper;
+import xyz.wagyourtail.jsmacros.client.api.classes.Draw2D;
+import xyz.wagyourtail.jsmacros.client.api.helpers.ClickableWidgetHelper;
 import xyz.wagyourtail.jsmacros.client.api.helpers.ItemStackHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.CheckBoxWidgetHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.CyclingButtonWidgetHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.LockButtonWidgetHelper;
+import xyz.wagyourtail.jsmacros.client.api.helpers.SliderWidgetHelper;
 import xyz.wagyourtail.jsmacros.client.api.helpers.TextFieldWidgetHelper;
 import xyz.wagyourtail.jsmacros.client.api.helpers.TextHelper;
 import xyz.wagyourtail.jsmacros.client.api.sharedclasses.PositionCommon;
 import xyz.wagyourtail.jsmacros.client.api.sharedclasses.PositionCommon.Pos2D;
 import xyz.wagyourtail.jsmacros.client.api.sharedclasses.PositionCommon.Vec2D;
 import xyz.wagyourtail.jsmacros.client.api.sharedclasses.RenderCommon;
+import xyz.wagyourtail.jsmacros.client.api.sharedclasses.RenderCommon.*;
 import xyz.wagyourtail.jsmacros.client.api.sharedinterfaces.IDraw2D;
 import xyz.wagyourtail.jsmacros.client.api.sharedinterfaces.IScreen;
 import xyz.wagyourtail.jsmacros.core.Core;
 import xyz.wagyourtail.jsmacros.core.MethodWrapper;
+import xyz.wagyourtail.wagyourgui.elements.CheckBox;
+import xyz.wagyourtail.wagyourgui.elements.Slider;
 
 import java.util.*;
 import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.BooleanSupplier;
 
 @Mixin(Screen.class)
 @Implements(@Interface(iface = IScreen.class, prefix = "soft$"))
 public abstract class MixinScreen extends AbstractParentElement implements IScreen, IScreenInternal {
```

Comment:
should we concider some kind of wrapper for normal screens, so that we don't need to do all this in a mixin?

Reply:
Due to the structure of this mod a wrapper would be much more fitting and may prove useful in the future for specialized screen helpers (the book screen for example). Except for the `add` methods and a few others, most methods must remain in the Mixin to function properly, but it would still make it more clear which functions are part of the public API, as many have tried to use the `render` method for example.
That being said, I think this would be a step in the right direction and could be implemented in a (smaller or larger) refactor for version 1.9.0.

Reply:
yeah. leave it for the future